controllers/spaceprovisionerconfig/spaceprovisionerconfig_controller.go

```diff
@@ -184,7 +184,8 @@ func (r *Reconciler) determineCapacityReadyState(spc *toolchainv1alpha1.SpacePro
 // It always knows this fact so returning a bool is ok, in contrast to determinMemoryUtilizationReadyState.
 func determineSpaceCountReadyState(spc *toolchainv1alpha1.SpaceProvisionerConfig) bool {
 	max := spc.Spec.CapacityThresholds.MaxNumberOfSpaces
-	return max == 0 || max > uint(spc.Status.ConsumedCapacity.SpaceCount)
+	//the value of this is not going beyond 100 and it won't overflow, hence its okay to ignore the overflow linter error
+	return max == 0 || max > uint(spc.Status.ConsumedCapacity.SpaceCount) // nolint:gosec
```

Comment:
the comment is not true for this case - the line is casting `spc.Status.ConsumedCapacity.SpaceCount` which should be `0` and more, it shouldn't ever go under `0`. 
The second option would be changing the comparison to:
```go
return max == 0 || int(max) > spc.Status.ConsumedCapacity.SpaceCount
```
stating that we don't expect that the max number of Space count would ever go above the max size of int